x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/string/Concat.java

```diff
@@ -83,13 +89,29 @@ public ExpressionEvaluator.Factory toEvaluator(Function<Expression, ExpressionEv
 
     @Evaluator
     static BytesRef process(@Fixed(includeInToString = false) BreakingBytesRefBuilder scratch, BytesRef[] values) {
+        ensureIsBelowMaxLength(values);
         scratch.clear();
         for (int i = 0; i < values.length; i++) {
             scratch.append(values[i]);
         }
         return scratch.bytesRefView();
     }
 
+    private static void ensureIsBelowMaxLength(BytesRef[] values) {
+        long length = 0;
+        for (var v : values) {
+            length += v.length;
+        }
+        if (length > MAX_CONCAT_LENGTH) {
+            throw new EsqlClientException("concatenating more than [" + MAX_CONCAT_LENGTH + "] bytes is not supported") {
```

Comment:
thought: in my previous PRs, I used `VerificationException` instead of an anonymous `EsqlClientException`, so that we don't send back an exception that reads `type: ""` in the REST response.

Reply:
`EsqlClientException.status()` returns `INTERNAL_SERVER_ERROR`, which is a 500 error. I thought of keeping the 4xx range.

Reply:
EsqlClientException.status() returns a `INTERNAL_SERVER_ERROR`, which is a 500 error. I wanted to keep the 4xx range.
`VerificationException` does that too, but it seems a dedicated exception.